### DISCUSSION TITLE
nginx upstream_response_time confict with skywalking shows

### DISCUSSION THREAD

**Discussion by guangdashao** [apache/skywalking#8182] [Q&A]
Nginx LUA Agent worked with  OpenResty

Nginx LUA Agent: 0.50
OpenResty: 1.17.8.1

in the browser , i found "http://skywalking.asgtjf.com.cn/supv/v1/0/aa5organizations?_=1637823163483"takes 1.58s

<img width="937" alt="afd7a6073cd668a6f9cea9ee91b7ef4" src="https://user-images.githubusercontent.com/54162228/143396002-4a17afba-0158-44e4-b466-e3962f14bbe5.png">

and in nginx logs, the request's request_time:1.570  ,upstream_response_time: 1.571

   10.1.161.12 - - [25/Nov/2021:15:06:42 +0800] "GET /supv/v1/0/aa5organizations?_=1637823163483 HTTP/1.1" 200 13135 "http://skywalking.asgtjf.com.cn/work.html" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/96.0.4664.45 Safari/537.36" "-"1.570 1.571

then  in skywalking trace ui ,i found like this

<img width="953" alt="cf7306b6e18b83fcce6d3e769976e76" src="https://user-images.githubusercontent.com/54162228/143396078-7f41a0bd-8192-4cb1-b841-275a6b22e80a.png">

skywalking told me : the request total duration is 1570ms, but the maximum duration (1512ms) is in nginx。

I'm confused， who was it right， nginx's log or skywalking?

**Comment by wu-sheng** [comment]
It is common to have tiny difference. Different mechanisms calculate duration from different ways.
Such as browser at least include network cost, and browser unpackaging HTTP.
This is why at topology, you can see detect points of server and client sides.
